### PR TITLE
Port of #3822: Add Support for FLOPS Calculation in `WandbCallback` for `keras`

### PR DESCRIPTION
**Pull request by dmitryduev**
Description
-----------
See https://github.com/wandb/client/pull/3822

Testing
-------
@ayulockin added functional tests.

Checklist
-------
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

File: wandb/integration/keras/keras.py [hunk 1]
```diff
@@ -30,6 +30,16 @@ def _check_keras_version():
         )
 
 
+def _can_compute_flops() -> bool:
+    """
+    FLOPS computation is restricted to TF 2.x as it requires tf.compat.v1
+    """
+    if parse_version(tf.__version__) >= parse_version("2.0.0"):
```
Comment (kptkin) [inline wandb/integration/keras/keras.py:37]: lol... do you want to put the import here ;)

File: wandb/integration/keras/keras.py [hunk 2]
```diff
@@ -66,12 +78,10 @@ def is_generator_like(data):
 
 
 def patch_tf_keras():
-
+    from pkg_resources import parse_version
     from tensorflow.python.eager import context
 
-    from tensorflow import __version__ as tf_version
```
Comment (kptkin) [inline wandb/integration/keras/keras.py:72]: i don't remember if there was a reason why we did it this way... but looks like it should be fine

File: tests/functional_tests/keras/keras_flops_functional_tf_2_6.yea [hunk 3]
```diff
@@ -0,0 +1,18 @@
+id: 0.keras.flopsfunctional.tf26
```
Comment (kptkin) [inline tests/functional_tests/keras/keras_flops_functional_tf_2_6.yea:1]: hmmm... do we care testing against earlier version of tf?

**Review by kptkin — APPROVED** [review]
🔥

File: tests/functional_tests/keras/keras_flops_functional_tf_2_6.yea (hunk 3 above)
Comment (dmitryduev) [reply tests/functional_tests/keras/keras_flops_functional_tf_2_6.yea:1]: prob not, it's been there (unchanged) since 2.0

Comment (dmitryduev) [reply tests/functional_tests/keras/keras_flops_functional_tf_2_6.yea:1]: what we should care about is testing more recent tf versions lol ;0

File: wandb/integration/keras/keras.py [hunk 4]
```diff
@@ -1000,3 +1022,53 @@ def _save_model_as_artifact(self, epoch):
...
+            )
+            .with_empty_output()
+            .build()
+        )
+
+        flops = tf.compat.v1.profiler.profile(
+            graph=frozen_func.graph, run_meta=run_meta, cmd="scope", options=opts
+        )
+
+        tf.compat.v1.reset_default_graph()
```
[the remark this answers is not in the document]
Comment (dmitryduev) [reply wandb/integration/keras/keras.py:1071]: @ayulockin would know better. my guess is this is necessary bc to compute FLOPS for a model, tf/tf.compat.v1.profiler needs to eval its graph on a single/few input batch(es), so once that's done, it's probably a good idea to reset things.

File: tests/functional_tests/keras/keras_flops_functional_tf_2_6.yea (hunk 3 above)
Comment (kptkin) [reply tests/functional_tests/keras/keras_flops_functional_tf_2_6.yea:1]: yeah... that's also a good point

**Pull request merged**